Event progress update uses userAchievements

dinoboff opened the pull request:
fixes #121

TODO:
- [x] update datastore service
- [x] fix ranking tab

dinoboff committed:
Event progress update uses userAchievements

Also remove “classMentors/eventRankings” related methods

dinoboff commented on src/classmentors/services.js:
```diff
@@ -1349,11 +1350,11 @@ export function clmDataStoreFactory(
+
+        Object.keys(data.userAchievements || {}).forEach(serviceId => {
+          ranking[serviceId] = data.userAchievements[serviceId].totalAchievements || 0;
+        });
```
Just realised ranking is not pre-calculated anymore. I will just remove that part.

dinoboff committed:
Update event ranking view and its controller

dinoboff commented:
@SingaporeClouds Event view and services are updated to use userAchievements.

Note that sorting still use angular orderBy filter but happens in the controller so that sorting only happens when sorting key/order changes, not for every scope digest cycle.

Things that didn't seems to be implemented (and are not implemented in this PR either):
- update ranking on participants profile/userAchievements update.
- rank in school for school event.
- ranked service (a way to edit the list of service to include in the ranking).
- paging.

Did I miss any of them? Should I implement them in this PR?

dinoboff commented:
@SingaporeClouds rebased and removed left over testing code (I was arbitrary changing total count to have something to sort).
@shiquan-lim Please review and check if I missed a feature that currently supported. Also, are there other views needing updated

SingaporeClouds commented:
The event rankings appear to break in a few places with this PR. There are numbers being concatenated as strings rather than added as integers and there are some "<!" symbols appearing that do not happen prior to this PR. 

Also, there appears to be a mixture of white and grey on the profile pages that I am not sure was intentional or not. Pictures attached. 

![image](https://cloud.githubusercontent.com/assets/116418/19112554/0f5b2be6-8b38-11e6-8704-f016b8a597ca.png)

![image](https://cloud.githubusercontent.com/assets/116418/19112653/9b29295c-8b38-11e6-954a-98be7ca12a34.png)

dinoboff commented:
I will fix those.

There's also an issue with the refresh all button that I will fix too.

For the profile page, it's unrelated. I will fix it in an other PR.

dinoboff committed:
Save service details with user participations

Allow owner to request update of the participant achievements without querying all profile.

It also fix updating participants user details (display name, school or gravatar).

dinoboff commented:
Rebased

@SingaporeClouds Can you please review again.
